File: shimmer-android-example/src/main/java/com/facebook/shimmer/example/MainActivity.java
/*
 * This file provided by Facebook is for non-commercial testing and evaluation purposes only.
 * Facebook reserves all rights not expressly granted.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
 * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package com.facebook.shimmer.example;

import android.animation.ObjectAnimator;
import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;
import com.facebook.shimmer.ShimmerFrameLayout;

public class MainActivity extends Activity {

  private ShimmerFrameLayout mShimmerViewContainer;
  private Button[] mPresetButtons;
  private int mCurrentPreset = -1;
  private Toast mPresetToast;

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    setContentView(R.layout.main);
    mShimmerViewContainer = (ShimmerFrameLayout) findViewById(R.id.shimmer_view_container);

    mPresetButtons = new Button[]{
        (Button) findViewById(R.id.preset_button0),
        (Button) findViewById(R.id.preset_button1),
        (Button) findViewById(R.id.preset_button2),
        (Button) findViewById(R.id.preset_button3),
        (Button) findViewById(R.id.preset_button4),
    };
    for (int i = 0; i < mPresetButtons.length; i++) {
      final int preset = i;
      mPresetButtons[i].setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View view) {
          selectPreset(preset, true);
        }
      });
    }
  }

  @Override
  protected void onStart() {
    super.onStart();

    selectPreset(0, false);
  }

  @Override
  public void onResume() {
    super.onResume();
    mShimmerViewContainer.startShimmerAnimation();
  }

  @Override
  public void onPause() {
    mShimmerViewContainer.stopShimmerAnimation();
    super.onPause();
  }

  /**
   * Select one of the shimmer animation presets.
   *
   * @param preset index of the shimmer animation preset
   * @param showToast whether to show a toast describing the preset, or not
   */
  private void selectPreset(int preset, boolean showToast) {
    if (mCurrentPreset == preset) {
      return;
    }
    if (mCurrentPreset >= 0) {
      mPresetButtons[mCurrentPreset].setBackgroundResource(R.color.preset_button_background);
    }
    mCurrentPreset = preset;
    mPresetButtons[mCurrentPreset].setBackgroundResource(R.color.preset_button_background_selected);

    // Save the state of the animation
    boolean isPlaying = mShimmerViewContainer.isAnimationStarted();

    // Reset all parameters of the shimmer animation
    mShimmerViewContainer.useDefaults();

    // If a toast is already showing, hide it
    if (mPresetToast != null) {
      mPresetToast.cancel();
    }

    switch (preset) {
      default:
      case 0:
        // Default
        mPresetToast = Toast.makeText(this, "Default", Toast.LENGTH_SHORT);
        break;
      case 1:
        // Slow and reverse
        mShimmerViewContainer.setDuration(5000);
        mShimmerViewContainer.setRepeatMode(ObjectAnimator.REVERSE);
        mPresetToast = Toast.makeText(this, "Slow and reverse", Toast.LENGTH_SHORT);
        break;
      case 2:
        // Thin, straight and transparent
        mShimmerViewContainer.setBaseAlpha(0.1f);
        mShimmerViewContainer.setDropoff(0.1f);
        mShimmerViewContainer.setTilt(0);
        mPresetToast = Toast.makeText(this, "Thin, straight and transparent", Toast.LENGTH_SHORT);
        break;
      case 3:
        // Sweep angle 90
        mShimmerViewContainer.setAngle(ShimmerFrameLayout.MaskAngle.CW_90);
        mPresetToast = Toast.makeText(this, "Sweep angle 90", Toast.LENGTH_SHORT);
        break;
      case 4:
        // Spotlight
        mShimmerViewContainer.setBaseAlpha(0);
        mShimmerViewContainer.setDuration(2000);
        mShimmerViewContainer.setDropoff(0.1f);
        mShimmerViewContainer.setIntensity(0.35f);
        mShimmerViewContainer.setMaskShape(ShimmerFrameLayout.MaskShape.RADIAL);
        mPresetToast = Toast.makeText(this, "Spotlight", Toast.LENGTH_SHORT);
        break;
    }

    // Show toast describing the chosen preset, if necessary
    if (showToast) {
      mPresetToast.show();
    }

    // Setting a value on the shimmer layout stops the animation. Restart it, if necessary.
    if (isPlaying) {
      mShimmerViewContainer.startShimmerAnimation();
    }
  }
}

File: sample/src/main/java/com/facebook/shimmer/sample/MainActivity.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.shimmer.sample

import android.animation.ValueAnimator
import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import com.facebook.shimmer.Shimmer
import com.facebook.shimmer.ShimmerFrameLayout
import kotlinx.android.synthetic.main.main.*

class MainActivity : Activity(), View.OnClickListener {
  private lateinit var shimmerViewContainer: ShimmerFrameLayout
  private lateinit var presetButtons: Array<Button>
  private var currentPreset = -1
  private var toast: Toast? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.main)
    shimmerViewContainer = shimmer_view_container
    presetButtons =
        arrayOf(
            preset_button0,
            preset_button1,
            preset_button2,
            preset_button3,
            preset_button4,
            preset_button5,
            preset_button6)
    presetButtons.forEach { it.setOnClickListener(this@MainActivity) }
    selectPreset(0, false)
  }

  override fun onClick(v: View) {
    selectPreset(presetButtons.indexOf(v as Button), true)
  }

  public override fun onResume() {
    super.onResume()
    shimmerViewContainer.startShimmer()
  }

  public override fun onPause() {
    shimmerViewContainer.stopShimmer()
    super.onPause()
  }

  private fun selectPreset(preset: Int, showToast: Boolean) {
    if (currentPreset == preset) {
      return
    }

    if (currentPreset >= 0) {
      presetButtons[currentPreset].setBackgroundResource(R.color.preset_button_background)
    }
    currentPreset = preset
    presetButtons[currentPreset].setBackgroundResource(R.color.preset_button_background_selected)

    // If a toast is already showing, hide it
    toast?.cancel()

    val shimmerBuilder = Shimmer.AlphaHighlightBuilder()
    shimmerViewContainer.setShimmer(
        when (preset) {
          1 -> {
            // Slow and reverse
            toast = Toast.makeText(this, "Slow and reverse", Toast.LENGTH_SHORT)
            shimmerBuilder.setDuration(5000L).setRepeatMode(ValueAnimator.REVERSE)
          }
          2 -> {
            // Thin, straight and transparent
            toast = Toast.makeText(this, "Thin, straight and transparent", Toast.LENGTH_SHORT)
            shimmerBuilder.setBaseAlpha(0.1f).setDropoff(0.1f).setTilt(0f)
          }
          3 -> {
            // Sweep angle 90
            toast = Toast.makeText(this, "Sweep angle 90", Toast.LENGTH_SHORT)
            shimmerBuilder.setDirection(Shimmer.Direction.TOP_TO_BOTTOM).setTilt(0f)
          }
          4 -> {
            // Spotlight
            toast = Toast.makeText(this, "Spotlight", Toast.LENGTH_SHORT)
            shimmerBuilder
                .setBaseAlpha(0f)
                .setDuration(2000L)
                .setDropoff(0.1f)
                .setIntensity(0.35f)
                .setShape(Shimmer.Shape.RADIAL)
          }
          5 -> {
            // Spotlight angle 45
            toast = Toast.makeText(this, "Spotlight angle 45", Toast.LENGTH_SHORT)
            shimmerBuilder
                .setBaseAlpha(0f)
                .setDuration(2000L)
                .setDropoff(0.1f)
                .setIntensity(0.35f)
                .setTilt(45f)
                .setShape(Shimmer.Shape.RADIAL)
          }
          6 -> {
            // Off
            toast = Toast.makeText(this, "Off", Toast.LENGTH_SHORT)
            null
          }
          else -> {
            toast = Toast.makeText(this, "Default", Toast.LENGTH_SHORT)
            shimmerBuilder
          }
        }?.build())

    // Show toast describing the chosen preset, if necessary
    if (showToast) {
      toast?.show()
    }
  }
}
